### PR TITLE
Structured Logging

### PR DESCRIPTION
**Pull request by pbrisbin**
Replaces RIO's `HasLogFunc` with Blammo. This also removed all our own log formatting and coloring-related code for Blammo's. 

This results in our logs being structured:

stackctl-blammo_000

Other user-facing changes:

- `-v` can be passed more than once, to increase from App DEBUG to App+Amazonka DEBUG to App+Amazonka TRACE
- The format of `version` changed slightly, from simply `{version}` to `Stackctl v{version}`

Architectural changes:

I moved the fetching of the current AWS Account/Region (called `AwsScope`) to app initialization and made it available anywhere by the `HasAwsScope` class. This is more efficient and allows us to add this information via `withThreadContext` around most logging, rather than in specific places, only once we'd fetched it.

At this point, the only reason for `RIO` is the base of our `Prelude` module. I plan to remove that as well, possibly for `Relude`.

~~**TODO**: I think I want to integrate this version in PlatformCLI and run its integration tests before merging.~~ PlatformCLI's Integration tests look good!

### CODE REVIEW

**Comment by pbrisbin** [comment]
I'll follow up with documentation updates after.

**Comment by pbrisbin** [comment]
What it looks like over in PlatformCLI,



Kind of neat (and also kind of the point) that the `withThreadContext` I call in PlatformCLI merges with the ones call in Stackctl to make for fully context-aware lines like that. :ok_hand:

File: src/Stackctl/AWS/Core.hs [hunk 1]
```diff
@@ -40,28 +38,42 @@ newtype AwsEnv = AwsEnv
...
+  env <- liftIO $ newEnv discover
+  AwsEnv <$> configureLogging env
+
+configureLogging :: MonadLoggerIO m => Env -> m Env
+configureLogging env = do
+  loggerIO <- askLoggerIO
+  pure $ env
+    { AWS.envLogger = \level msg -> do
+      loggerIO
+        defaultLoc -- TODO: there may be a way to get a CallStack/Loc
```
Comment (danroyo) [inline src/Stackctl/AWS/Core.hs:52]: this would be neat if it pans out

**Review by danroyo — APPROVED** [review]
awesome to see this getting integrated into so many projects so quickly

**Pull request merged**